library/log_validation.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+include_once("../interface/globals.php");
+include_once("$srcdir/log.inc");
+include_once("$srcdir/formdata.inc.php");
+require_once("$srcdir/formatting.inc.php");
+
+
+	$valid  = true;
+	$errors = array(); 
+	catch_logs();
+	$sql = sqlStatementNoLog("select *from log_validator");
+	while($row = sqlFetchArray($sql)){
+		$logEntry = sqlQuery("select * from log where id = ".$row['log_id']);
+		if(empty($logEntry)){
+			$valid = false;
+			array_push($errors, "Audit log entry #".$row['log_id']." is missing");
+		}
+		else if($row['log_checksum'] != $logEntry['checksum']){
+			$valid = false;
+			array_push($errors,"Audit log tampering evident at entry #".$row['log_id']);
```

Comment:
same as above to make more clear (do not use # sign):
``````
array_push($errors, xl("Audit log tampering evident at entry number") . " " . $row['log_id']);